app/Command/Ide.hs

```diff
@@ -164,15 +164,16 @@ startServer port env = withSocketsDo $ do
           case decodeT cmd of
             Just cmd' -> do
               let message duration =
-                    "Command " <> commandName cmd'
-                    <> " took "
-                    <> displayTimeSpec duration
-              result <- logPerf message (runExceptT (handleCommand cmd'))
-              -- $(logDebug) ("Answer was: " <> T.pack (show result))
+                    "Command "
+                      <> commandName cmd'
+                      <> " took "
+                      <> displayTimeSpec duration
+              logPerf message $ do
+                result <- runExceptT (handleCommand cmd')
+                case result of
+                  Right r  -> liftIO $ catchGoneHandle (BSL8.hPutStrLn h (Aeson.encode r))
+                  Left err -> liftIO $ catchGoneHandle (BSL8.hPutStrLn h (Aeson.encode err))
```

Comment:
Needs to be explicitly annotated with `forall` so I didn't do it so far, since it ends up being more code.